### PR TITLE
Sbt Plugin Tests

### PR DESCRIPTION
**Pull request by sokrahta**
discovered sbt scripted testing framework, added tests for all the present plugins.

Sbt Plugin Tests
------------------------
1. run ```sbt scripted```
1. alternatively open ```sbt``` shell and run any suite(s) interactively
  * execute ```scripted PluginTests/HelloWorld```
  * execute ```scripted PluginTests/CoreSettings```
  * execute ```scripted PluginTests/Style```
  * execute ```scripted PluginTests/Coverage```
1. look for a **happy green success** at the end of each
  * there may be nasty red errors inside some build tests, by design!

[Review on Reviewable](https://reviewable.io/reviews/socrata/socrata-sbt-plugins/4)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -3,16 +3,16 @@ organization := "com.socrata"
 scalaVersion in Global := "2.10.4"
 sbtPlugin := true
 
-resolvers ++= Seq(
-  "socrata release"   at "https://repository-socrata-oss.forge.cloudbees.com/release",
-//  "socrata snapshot"  at "https://repository-socrata-oss.forge.cloudbees.com/snapshot",
-  "sonatype release"  at "https://oss.sonatype.org/content/repositories/releases",
-//  "sonatype snapshot" at "https://oss.sonatype.org/content/repositories/snapshots",
+resolvers ++= Seq(Classpaths.sbtPluginReleases, Resolver.mavenLocal,
   "thricejamie bintray" at "http://dl.bintray.com/thricemamie/sbt-plugins",
```
Comment (urmilan) [inline build.sbt]: s/mamie/jamie ??

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,13 +1,13 @@
-resolvers ++= Seq(
-  "socrata release"   at "https://repository-socrata-oss.forge.cloudbees.com/release",
-//  "socrata snapshot"  at "https://repository-socrata-oss.forge.cloudbees.com/snapshot",
-  "sonatype release"  at "https://oss.sonatype.org/content/repositories/releases",
-//  "sonatype snapshot" at "https://oss.sonatype.org/content/repositories/snapshots",
+resolvers ++= Seq(Classpaths.sbtPluginReleases, Resolver.mavenLocal,
   "thricejamie bintray" at "http://dl.bintray.com/thricemamie/sbt-plugins",
```
Comment (urmilan) [inline project/plugins.sbt]: s/mamie/jamie

File: build.sbt (hunk 1 above)
Comment (urmilan) [inline build.sbt]: Actually I had to change it to this for the artifact to download successfully: 
Resolver.url("thricejamie bintray", url("http://dl.bintray.com/thricejamie/sbt-plugins"))(Resolver.ivyStylePatterns)

File: project/plugins.sbt (hunk 2 above)
Comment (urmilan) [inline project/plugins.sbt]: See above

**Comment by urmilan** [comment]
Annoying, it seems some of the other plugins haven't upgraded to the latest version of SBT yet, so we're getting these eviction warnings:
org.scala-sbt:sbt:(0.13.0, 0.13.0-RC3) -> 0.13.7 (caller: com.eed3si9n:sbt-assembly:0.9.1 (sbtVersion=0.13, scalaVersion=2.10), com.typesafe:sbt-mima-plugin:0.1.6 (sbtVersion=0.13, scalaVersion=2.10), com.github.gseitz:sbt-release:0.8.5 (sbtVersion=0.13, scalaVersion=2.10), org.scala-sbt:scripted-plugin:0.13.7)

**Comment by urmilan** [comment]
Do we still actually need the HelloWorld plugin anymore?

**Comment by urmilan** [comment]
:+1: I'm not sure what the best practice is wrt suppressing eviction warnings but apart from that - LGTM

**Comment by sokrahta** [comment]
plan to remove the helloworld and meow plugins before releasing version 0.1.0

**Comment by sokrahta** [comment]
oh yeah, the eviction warnings are from some transitive dependencies in cloudbees, will address those plugins in the next pull request. 
sbt-assembly:0.9.1 -> 0.12.0
sbt-mima-plugin:0.1.6 -> ?
sbt-release:0.8.5 -> ?

**Comment by velvia** [comment]
Good work guys.  Looking forward to trying this out.

On Thu, Jan 15, 2015 at 11:33 AM, John Krah <notifications@github.com>
wrote:

> oh yeah, the eviction warnings are from some transitive dependencies in
> cloudbees, will address those plugins in the next pull request.
> sbt-assembly:0.9.1 -> 0.12.0
> sbt-mima-plugin:0.1.6 -> ?
> sbt-release:0.8.5 -> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/socrata/socrata-sbt-plugins/pull/4#issuecomment-70146669>
> .
>



-- 
Evan Chan   |   Principal Engineer, Socrata     |    @evanfchan
*Socrata <http://www.socrata.com/> *| The Data Experience Company™
*Socrata isn't about worldly success, but about changing the world
successfully. This is why I'm excited to be here.*
*What does Socrata do? <https://www.youtube.com/watch?v=yyz86MlkriQ> *(watch
our 2 min video)

**Comment by sokrahta** [comment]
Jenkins build of this branch succeeded
https://socrata-oss.ci.cloudbees.com/job/socrata-sbt-plugins/2/

**Comment by urmilan** [comment]
:+1: Feel free to merge.
One small note, you'll need to add a publish step to your cloudbees job so that other projects can pull down the plugin jar.

**Pull request merged**